tests/test_squash_command.rs

```diff
@@ -312,6 +312,46 @@ fn test_squash_description() {
     "###);
 }
 
+#[test]
+fn test_squash_empty() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "parent"]);
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+
+    let stdout = test_env.jj_cmd_success(&repo_path, &["squash"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Working copy now at: 300625b8982b (no description set)
+    "###);
+    insta::assert_snapshot!(get_description(&test_env, &repo_path, "@-"), @r###"
+    parent
+    "###);
+
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "child"]);
+    let edit_script = test_env.set_up_fake_editor();
+    std::fs::write(&edit_script, "dump editor0").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["squash"]);
+    insta::assert_snapshot!(get_description(&test_env, &repo_path, "@-"), @r###"
+    parent
+
+    child
+    "###);
+
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor0")).unwrap(), @r###"
```

Comment:
I think we have enough tests of the text we pass to the editor, so feel free to drop this (and the `dump editor0` line above) and just leave the test of the resulting description above.

Reply:
It seems that without 
```
let edit_script = test_env.set_up_fake_editor();
std::fs::write(&edit_script, "dump editor0").unwrap();
```
The following line fails `test_env.jj_cmd_success(&repo_path, &["squash"]);`:

```
Unexpected failure.
code-1
stderr=``````
Standard input is not a terminal
Error: Editor \'pico\' exited with an error
...
```

So I am not sure how to silence it. Yet.

Reply:
Also, I probably just want to assert here that the user was asked to merge the messages, rather than the actual text

Reply:
I think it should still work if you just removed the second of those two lines, i.e. removing the "dump editor0" line but keeping the `set_up_fake_editor()` line.

Reply:
This did not work. I have finished for today, will try again tomorrow, if I can't find a solution - I will submit as is.

Reply:
That was arguably a bug in the test environment and should be fixed by #1127

Reply:
Thanks, it worked! So I removed all unnecessary editor setups/checks in the test.